core/shared/src/main/scala/zio/Promise.scala

```diff
@@ -70,13 +70,19 @@ class Promise[E, A] private (private val state: AtomicReference[State[E, A]]) ex
    * Kills the promise with the specified error, which will be propagated to all
    * fibers waiting on the value of the promise.
    */
-  final def die(e: Throwable): UIO[Boolean] = done(IO.die(e))
+  final def die(e: Throwable): UIO[Boolean] = completeWith(IO.die(e))
+
+  /**
+   * Exits the promise with the specified exit, which will be propagated to all
+   * fibers waiting on the value of the promise.
+   */
+  final def done(e: Exit[E, A]): UIO[Boolean] = completeWith(IO.done(e))
 
   /**
    * Completes the promise with the specified result. If the specified promise
    * has already been completed, the method will produce false.
    */
-  final def done(io: IO[E, A]): UIO[Boolean] =
+  final def completeWith(io: IO[E, A]): UIO[Boolean] =
```

Comment:
I'd just call this `complete`. Also it would be nice to keep the old `done` (an overloaded method, as an alias) with a deprecation warning. Just to give people a chance to migrate. We can delete it on 1.0.

Reply:
@jdegoes Sure, done it.